oxd-server/src/main/java/org/gluu/oxd/server/persistence/JDBCPersistenceProvider.java

```diff
@@ -0,0 +1,84 @@
+package org.gluu.oxd.server.persistence;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Strings;
+import org.apache.commons.dbcp.BasicDataSource;
+import org.gluu.oxd.common.ErrorResponseCode;
+import org.gluu.oxd.common.Jackson2;
+import org.gluu.oxd.server.HttpException;
+import org.gluu.oxd.server.OxdServerConfiguration;
+import org.gluu.oxd.server.service.ConfigurationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public class JDBCPersistenceProvider implements SqlPersistenceProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JDBCPersistenceProvider.class);
+
+    private ConfigurationService configurationService;
+    private static BasicDataSource dataSource = null;
+
+    public JDBCPersistenceProvider(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    @Override
+    public void onCreate() {
+        JDBCConfiguration jdbcConfiguration = asJDBCConfiguration(configurationService.getConfiguration());
+        validate(jdbcConfiguration);
+        if (dataSource == null) {
```

Comment:
`onCreate()` is called one time on start up thus it's always null. If we got `onCreate` called second time then it's bug which must be fixed. At the end `if` here is redundant.

Reply:
Removed `if`. `Create` is not getting called twice.